include/spore/codegen/codegen_app.hpp

```diff
@@ -89,8 +89,32 @@ namespace spore::codegen
         {
             std::vector<std::filesystem::path> inputs = glob::rglob(step.input);
 
+            // search for templates starting at the working directory and then going
+            // through configured template prefix paths.
+            std::vector<std::string> templates;
```

Comment:
I have a large repo with dozens of subprojects that have a code generation step.

The generation is done via the `spore_codegen` cmake integration and the `codegen.json` configuration file lives in the subproject tree, so they are all over the repo.

The thing is they all share the same template and as it stands the app will only search for templates relative to the WORKING_DIRECTORY.

Suppose I have the following structure:

```
msi@DESKTOP-JP2BUFD:~/tmp/codegen$ tree
├── codegen
│   └── templates
│       └── template.inja
└── components
    ├── comp_a
    │   └── libs
    │       └── project_a
    │           ├── CMakeLists.txt
    │           └── codegen.json
 ```

Inside the project_a/CMakeLists.txt is the call to `spore_codegen`. 

This means from inside `components/comp_a/libs/project_a/codegen.json` I would have to set the template path as `../../../codegen/templates/template.inja'`.

This causes problems if I ever move the template, as I would need to updated dozens of codegen.json files and also the `../../` thing just feels wrong.

So what I can do with this PR is pass the `TEMPLATE_PATHS "${CMAKE_SOURCE_DIR}/codegen/templates` option to `spore_codegen` and in the `codegen.json` just use `"templates": ["template.inja"]`

Reply:
I wonder if we should perhaps drop the `codegen.json` file altogether, and supply inputs and templates via the command line directly. Do you think this would give more flexibility? We could inject whatever templates and inputs via the CMake script with regards to all the predefined variables. If we want multiple steps, we could setup `spore_codegen` twice.